packages/tkg-clusterclass/bundle/config/packageinstalls/tkg-infra-clusterclass-pi.yaml

```
@@ -72,3 +84,135 @@ metadata:
+              exemptions:
+                type: object
+                description: "excemption configuration for the PodSecurityConfiguration."
+                properties:
+                  namespaces:
+                    type: array
+                    items:
+                      type: string
+                    description: "namespaces excluded to apply PodSecurityConfiguration Admission."
+      #@overlay/match missing_ok=True
+      patches:
+      #@overlay/remove
+      #@overlay/match by=overlay.map_key("name"),missing_ok=True
+      - name: podSecurityStandard
+      #@overlay/append
+      - name: podSecurityStandard
+        description: "Adds an admission configuration for PodSecurity to the kube-apiserver."
+        definitions:
+        - selector:
+            apiVersion: controlplane.cluster.x-k8s.io/v1beta1
+            kind: KubeadmControlPlaneTemplate
+            matchResources:
+              controlPlane: true
+          jsonPatches:
+          - op: add
+            path: "/spec/template/spec/kubeadmConfigSpec/clusterConfiguration/apiServer/extraArgs/admission-control-config-file"
+            value: "/etc/kubernetes/kube-apiserver-admission-pss.yaml"
+          - op: add
+            path: "/spec/template/spec/kubeadmConfigSpec/clusterConfiguration/apiServer/extraVolumes/-"
+            value:
+              name: admission-pss
+              hostPath: /etc/kubernetes/kube-apiserver-admission-pss.yaml
+              mountPath: /etc/kubernetes/kube-apiserver-admission-pss.yaml
+              readOnly: true
+              pathType: "File"
+          - op: add
+            path: "/spec/template/spec/kubeadmConfigSpec/files/-"
+            valueFrom:
+              template: |-
+                content: |-
+                  {{ $namespace_exemptions := "kube-system" }}
+                  {{ range ("(@= pod_security_namespace_exemptions @)" | split ",") }}
+                    {{ $namespace_exemptions := printf "%s, %q" $namespace_exemptions . }}
+                  {{ end }}
+                  {{ $defaultWarnAudit := "baseline" }}
+                  {{ if .podSecurityStandard.exemptions.namespaces }}
+                    {{ range $namespace := .podSecurityStandard.exemptions.namespaces }}
+                      {{ $namespace_exemptions = printf "%s, %q" $namespace_exemptions $namespace }}
+                    {{ end }}
+                  {{ end }}
+                  apiVersion: apiserver.config.k8s.io/v1
+                  kind: AdmissionConfiguration
+                  plugins:
+                  - name: PodSecurity
+                    configuration:
+                      apiVersion: pod-security.admission.config.k8s.io/v1beta1
+                      kind: PodSecurityConfiguration
+                      defaults:
+                        enforce: "{{ if .podSecurityStandard.enforce }}{{ .podSecurityStandard.enforce }}{{ end }}"
```

I might recommend something like the following
```suggestion
                        enforce: "{{ if .podSecurityStandard.enforce -}}
                            {{  .podSecurityStandard.enforce }}
                         {{- end }}"
```

For some of these it might make it easier to parse. You will need to use the proper `{{-` or `-}}` to make sure it is formatted correctly.